platform/mbed_application.c

```diff
@@ -69,17 +77,20 @@ static void powerdown_scb(uint32_t vtor)
     SCB->SCR = 0x00000000;
     // SCB->CCR     - Implementation defined value
     for (i = 0; i < 12; i++) {
-#if defined(__CORTEX_M7)
+#if defined(__CORTEX_M7) || defined(__CORTEX_M23)
```

Comment:
Not good - M23 has only 2 SHPR registers (the first is reserved).  Looks like we have mixed up the size of our register definitions... core_cm23.h is using uint32_t (x2) while core_cm7.h is using uint8_t (x12).  We need two for loops or should change SCB_Type in core_cm23.h.

Reply:
@TacoGrandeTX I fix it in https://github.com/ARMmbed/mbed-os/pull/6949/commits/3c14cb633222cbc42fc9c1ea257da49db3a73d15.